pkg/controller/control.go

```diff
@@ -656,6 +660,78 @@ func (c *Controller) checkUnmapMarkSnapChainRemoved() error {
 	return nil
 }
 
+func (c *Controller) SetMaximumSnapshotCount(count int) error {
+	c.Lock()
+	defer c.Unlock()
+
+	c.maximumSnapshotCount = count
```

Comment:
If we blindly set the c.maximumSnapshotCount to whatever the passed-in value `count` here, will it encounter this corner case:
1. c.maximumSnapshotCount is currently 10 and the current number of snapshot is 5
1. User set volume.Spec.SnapshotMaxCount to 2, we blindly set c.maximumSnapshotCount to 2
1. Now, the volume has more snapshots than c.maximumSnapshotCount which breaks the convention

IMO, we should failed to set the  c.maximumSnapshotCount to 2 while the current number of snapshots is already 5. WDYT?

Reply:
Yeah, we have related check in webhook. Do you think we also need to add it in longhorn-engine? Thanks.

https://github.com/longhorn/longhorn-manager/blob/4b5dbeca6ca731668d899492a8b11e878be6b47f/webhook/resources/volume/validator.go#L510-L524

Reply:
I think enforcing at the engine level will make it 100% sure. The webhook get data from CRs which might be out of date (we have been dealing with stale cache data issue for a while now in webhook)